src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala

```diff
@@ -182,6 +182,16 @@ object Hdfs {
   def globLines(p: Path, glob: String = "*"): Hdfs[Iterator[String]] =
     Hdfs.globFiles(p, glob).flatMap(_.map(Hdfs.readLines).sequenceU.map(_.toIterator.flatten))
 
+  /**
+   * @return glob string which will glob all the "leaves" files accessible under a given directory
+   */
+  def filesGlob(path: Path): Hdfs[Option[String]] =
+    globFilesRecursively(path).map { files =>
+      files.map(_.depth).maximum.map { max =>
+        Some(List.fill(max - path.depth)("*").mkString("/"))
+      }.getOrElse(None)
+    }
```

Comment:
Does this get all of the leaves, or only the leaves of the largest depth? eg. if you have `/dir/subdir1/file1` and `/dir/file2`, will this return `file1` and `file2` or just `file1`?